src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/LocalizedString.shared.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+#if !NETSTANDARD1_0
+	public class LocalizedString : INotifyPropertyChanged
+	{
+		public LocalizedString(Func<string> generator = null)
+            : this(LocalizationResourceManager.Current, generator)
+        {
+        }
+
+		public LocalizedString(LocalizationResourceManager localizationManager, Func<string> generator = null)
+		{
+			Generator = generator;
+			localizationManager.PropertyChanged += (sender, args) => Invalidate();
```

Comment:
I'm a little concerned about this event. This isn't a single class and the user can create a lot of instances of this (maybe without intention) and we never unsubscribe this event what can lead to a memory leak and odd behaviors. So I see two solutions for it:

1. Implement the IDisposable pattern;
2. Use the singleton pattern (as we did for `LocalizationResourceManager`), that way we will just one instance of it for all app lifecycle.

I would say that option 2 will be simple and enough for this scenario. Please let me know your thoughts.

Reply:
Hm, good point. 
1) `IDisposable` won't help if nobody would call `Dispose`
2) Yes, I think this is the way, but we need to be able to have different instances of `LocalizedString` with different generators. Maybe we can use `PropertyChanged` event from `LocalizationResourceManager`, but then `LocalizedString` must be immutable

Reply:
1. We can override the Destructor of the class and let the GC call the Dispose from there, but you know isn't too safe because we can't control the GC.
2. Why do we need different generators?

Reply:
1. Yes, this would work. I think this is the best option
2. In case strings will need to be changed. If generator is immutable the only way to change the string is to create a new instance. It's fine approach, just wanted to give more options. Yeah, changing generator would be conducing, better make it private

Reply:
Implemented `IDisposable`